scripts/west_commands/sign.py

```diff
@@ -633,3 +648,82 @@ def sign(self, command, build_dir, build_conf, formats):
+        siwx91x_encrypt_enable = getattr(args, 'siwx91x-encrypt-enable', build_conf.get('CONFIG_SIWX91X_SILABS_ENCRYPT', False))
+        siwx91x_sign_enable = getattr(args, 'siwx91x-sign-enable', build_conf.get('CONFIG_SIWX91X_SILABS_SIGN', False))
+        # Refer find_commandertool() function to identify the Commander tool.
+        commander_path = self.find_commandertool(command, args)
+        # Validate required settings
+        if not siwx91x_ota_encrypt_key:
+            command.die("SIWX91X_OTA_ENCRYPT_KEY not provided. Use --siwx91x-ota-encrypt-key=your_key or add CONFIG_SIWX91X_OTA_ENCRYPT_KEY in prj.conf.")
+        if not siwx91x_ota_sign_key:
+            command.die("SIWX91X_OTA_SIGN_KEY not provided. Use --siwx91x-ota-sign-key=/path/to/key or add CONFIG_SIWX91X_OTA_SIGN_KEY in prj.conf.")
+
+        # Validate private key path
+        if not os.path.isfile(siwx91x_ota_sign_key):
+            command.die(f"Private key not found at {siwx91x_ota_sign_key}. Please ensure the path is correct.")
+
+        return siwx91x_ota_encrypt_key, commander_path, siwx91x_ota_sign_key, siwx91x_mic_enable, siwx91x_encrypt_enable, siwx91x_sign_enable
+
+    def sign(self, command, build_dir, build_conf, formats):
+        """Sign the Zephyr binary using Silicon Labs Commander.
+commander_path
+        :param command: The Sign instance (provides args and utility methods)
+        :param build_dir: The build directory path
+        :param build_conf: BuildConfiguration object for the build directory
+        :param formats: List of formats to generate (e.g., ['bin', 'rps'])
+        """
+        self.command = command
+        args = command.args
+        b = pathlib.Path(build_dir)
+        kernel_name = build_conf.get('CONFIG_KERNEL_BIN_NAME')
+        # Setting the input and output paths
+        input_rps = b / 'zephyr' / f'{kernel_name}.bin.rps'
+        output_rps = args.sbin or (b / 'zephyr' / f'{kernel_name}_signed.bin.rps')
+        # Check if input binary exists
+        if not input_rps.is_file():
+            command.die(f"No .rps found at {input_rps}. Ensure the build generated kernel_name.bin.rps in {b / 'zephyr'}")
+
+        # Load configuration
+        siwx91x_ota_encrypt_key, commander_path, siwx91x_ota_sign_key, siwx91x_mic_enable, siwx91x_encrypt_enable, siwx91x_sign_enable = self.get_security_configs(build_conf, args, command)
+        siwx91x_ota_mic_key = siwx91x_ota_encrypt_key
+
+        sign_base = [
+            commander_path,
+            "rps",
+            "convert",
+            str(output_rps),
+            "--app", str(input_rps)
+        ]
+
+        if siwx91x_mic_enable:
+            sign_base.extend(["--mic", str(siwx91x_ota_mic_key)])
+
+        if siwx91x_encrypt_enable:
+            sign_base.extend(["--encrypt", str(siwx91x_ota_encrypt_key)])
+
+        if siwx91x_sign_enable:
+            sign_base.extend(["--sign", str(siwx91x_ota_sign_key)])
```

Comment:
Why is "OTA" everywhere in the key names but no "OTA" in the name of the booleans? This inconsistency gives the impression that there is some subtle difference between the keys and the booleans. But this code looks like there is none. Confusing.